src/components/SearchView.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { LibraryItem } from "./LibraryItem";
+import { LibraryView } from "../LibraryView";
+import { search, ItemData } from "../LibraryUtilities";
+
+interface SearchViewProps {
+    onSearchChanged: any;
```

Comment:
This should be renamed to `onSearchModeChanged`, and it would be nice if you can [give it a type](https://www.typescriptlang.org/docs/handbook/interfaces.html) like so:

```js
interface SearchModeChangedFunc {
    (inSearchMode: boolean): void;
}

interface SearchViewProps {
    onSearchModeChanged: SearchModeChangedFunc;
}
```